ui/components/app/connected-accounts-list/connected-accounts-list-item/connected-accounts-list-item.component.js

```diff
@@ -1,54 +1,91 @@
-    status: PropTypes.string,
-    action: PropTypes.node,
-    options: PropTypes.node,
-  };
-
-  static defaultProps = {
-    className: null,
-    options: null,
-    action: null,
-  };
-
-  render() {
-    const { address, className, name, status, action, options } = this.props;
-
-    return (
-      <div className={classnames('connected-accounts-list__row', className)}>
-        <div className="connected-accounts-list__row-content">
-          <Identicon
-            className="connected-accounts-list__identicon"
-            address={address}
-            diameter={32}
-          />
-          <div>
-            <p className="connected-accounts-list__account-name">
-              <strong>{name}</strong>
-            </p>
+export default function ConnectedAccountsListItem({
+  address,
+  className = null,
+  name,
+  status,
+  action = null,
+  options = null,
+  backgroundColor,
+}) {
+  const useBlockie = useSelector(getUseBlockie);
+  const containerbackgroundColor =
+    backgroundColor || BackgroundColor.backgroundDefault;
+  return (
+    <Box
+      className={classnames('connected-accounts-list__row', className)}
+      padding={4}
+      backgroundColor={containerbackgroundColor}
+      display={Display.Flex}
+      flexDirection={FlexDirection.Row}
+      justifyContent={JustifyContent.spaceBetween}
+      alignItems={AlignItems.center}
+    >
+      <Box className="connected-accounts-list__row-content" gap={4}>
+        <AvatarAccount
+          variant={
+            useBlockie
+              ? AvatarAccountVariant.Blockies
+              : AvatarAccountVariant.Jazzicon
+          }
+          address={address}
+          size={AvatarAccountSize.Md}
+        />
+        <Box display={Display.Flex} flexDirection={FlexDirection.Column}>
+          <Box>
+            <Text
+              variant={TextVariant.bodyLgMedium}
+              className="connected-accounts-list__account-name"
+            >
+              {name}
+            </Text>
             {status ? (
-              <p className="connected-accounts-list__account-status">
-                &nbsp;&nbsp;
+              <Text
+                variant={TextVariant.bodyMd}
+                className="connected-accounts-list__account-status"
+                color={TextColor.successDefault}
+              >
                 {status}
-              </p>
+              </Text>
             ) : null}
+          </Box>
+          <Box display={Display.Flex} flexDirection={FlexDirection.Column}>
             {action}
-          </div>
-        </div>
-        {options}
-      </div>
-    );
-  }
+          </Box>
+        </Box>
+      </Box>
+      {options}
+    </Box>
+  );
 }
+ConnectedAccountsListItem.propTypes = {
+  address: PropTypes.string.isRequired,
+  className: PropTypes.string,
+  name: PropTypes.node.isRequired,
+  status: PropTypes.string,
+  action: PropTypes.node,
+  options: PropTypes.node,
+  backgroundColor: PropTypes.string,
+};
```

Comment:
We should add documentation comments for each of these

Reply:
Updated here https://github.com/MetaMask/metamask-extension/pull/20811/commits/2ba5304bb2d3f6c43ae05731a3ed4d15c83fd963